packages/desktop-client/src/components/transactions/MobileTransaction.js

```diff
@@ -47,6 +81,50 @@ function getDescriptionPretty(transaction, payee, transferAcct) {
   return '';
 }
 
+function serializeTransaction(transaction, dateFormat) {
+  let { date, amount } = transaction;
+  return {
+    ...transaction,
+    date: formatDate(parseISO(date), dateFormat),
+    amount: integerToAmount(amount || 0),
+  };
+}
+
+function deserializeTransaction(transaction, originalTransaction, dateFormat) {
+  let { amount, date, ...realTransaction } = transaction;
+
+  let dayMonth = monthUtils.getDayMonthRegex(dateFormat);
+  if (dayMonth.test(date)) {
+    let test = parseDate(
+      date,
+      monthUtils.getDayMonthFormat(dateFormat),
+      new Date(),
+    );
+    if (isValidDate(test)) {
+      date = monthUtils.dayFromDate(test);
+    } else {
+      date = null;
+    }
+  } else {
+    let test = parseDate(date, dateFormat, new Date());
+    // This is a quick sanity check to make sure something invalid
+    // like "year 201" was entered
+    if (test.getFullYear() > 2000 && isValidDate(test)) {
+      date = monthUtils.dayFromDate(test);
+    } else {
+      date = null;
+    }
+  }
+
+  if (date == null) {
+    date =
+      (originalTransaction && originalTransaction.date) ||
+      monthUtils.currentDay();
+  }
+
+  return { ...realTransaction, date, amount: amountToInteger(amount || 0) };
+}
```

Comment:
Can you move these to a common file so they’re not duplicated with the desktop UI?

Reply:
Hmm, honestly they're different enough that I'd rather combine them as a follow-up in a focused PR where can carefully ensure nothing gets broken. As it is, swapping the mobile version of the desktop version breaks mobile, and vice versa breaks desktop.

Reply:
Good point! I didn’t notice that they were so different (and there are already 2 of them). Fine to keep them separate then.

Reply:
👍, will add this to the follow-up list